benches/barchart.rs

```diff
@@ -0,0 +1,63 @@
+use criterion::{criterion_group, criterion_main, Bencher, BenchmarkId, Criterion};
+use rand::Rng;
+use ratatui::{
+    buffer::Buffer,
+    layout::Rect,
+    widgets::{Bar, BarChart, BarGroup, Widget},
+};
+
+/// Benchmark for rendering a barchart.
+pub fn barchart(c: &mut Criterion) {
+    let mut group = c.benchmark_group("barchart");
+    let mut rng = rand::thread_rng();
+
+    for data_count in [64, 256, 2048] {
+        let data: Vec<Bar> = (0..data_count)
+            .map(|i| {
+                Bar::default()
+                    .label(format!("B{i}").into())
+                    .value(rng.gen_range(0..data_count))
+            })
+            .collect();
+
+        let bargroup = BarGroup::default().bars(&data);
+
+        // Render a basic barchart
+        group.bench_with_input(
+            BenchmarkId::new("render", data_count),
+            &BarChart::default().data(bargroup.clone()),
+            render,
+        );
+
+        // Render a barchart with multiple groups
+        group.bench_with_input(
+            BenchmarkId::new("render_grouped", data_count),
+            &BarChart::default()
+                // We call `data` multiple time to add multiple groups.
+                // This is not a duplicated call.
+                .data(bargroup.clone())
+                .data(bargroup.clone())
+                .data(bargroup.clone()),
```

Comment:
When I saw the result, I was expecting, that `barchart/render/64` and `barchart/render_grouped/64` contain both 64 bars for comparison. but the `render_grouped` has three time more bars. I think we should pass third of `bargroup` and not the full array

Reply:
Here 64 means **datasets** of 64 elements. But the meaning is not really important, the numbers are just to satisfy criterion which needs them to display the result as a table :

![image](https://github.com/ratatui-org/ratatui/assets/36198422/f5ff9e55-2198-445e-8038-563e274b3322)

I think passing a third of the array kind of defeats the purpose: you are then comparing the same thing with `barchart/render/64` and `barchart/render_grouped/64`.

This bench (imo) should be able to show any difference between using one or multiple datasets. In addition, I think it's good to mimic a "real world usage", thus, more datasets doesn't necessarily means less bar (in each dataset).

Hope this was clear enough, let me know your thoughts :)

Reply:
thanks for the explanation . just resolve all issues  :smile_cat: